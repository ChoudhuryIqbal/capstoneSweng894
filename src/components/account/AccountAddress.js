import React from 'react'
import { View, Platform } from 'react-native'
import { InputCustom, InputNumberCustom } from '../common'
import styles from '../styles/AccountAddress.styles'
import CustomInputStyles from '../styles/CustomInputStyles'

/**
 * Account Address components
 * @param {*} props 
 */
const AccountAddress = (props) => {
    return (
        <View>
            <View style={{flex: 1, flexDirection: 'row'}}>
                <InputCustom
                    placeholder="i.e. 100 Main Street"
                    label="Address: "
                    value={props.address}
                    onChangeText={props.onAddressChge}
                    error={props.errorAddress}
                    inputStyle = {CustomInputStyles.inputStyleOsLeft}
                    containerStyle = {CustomInputStyles.containerStyleLeft}      
                    labelStyle = {CustomInputStyles.labelStyle}
                    errorStyle = {Platform.Os === 'ios' ? CustomInputStyles.error : CustomInputStyles.errorAndroid} 
                />
                <InputCustom
                    placeholder="i.e. Malven"
                    label="City: "
                    value={props.city}
                    onChangeText={props.onCityChge}
                    error={props.errorCity}
                    inputStyle = {CustomInputStyles.inputStyleOsRight }
                    containerStyle = {CustomInputStyles.containerStyleRight}      
                    labelStyle = {CustomInputStyles.labelStyleRight}
                    errorStyle = {Platform.Os === 'ios' ? CustomInputStyles.error : CustomInputStyles.errorAndroid} 
                />
            </View>
             <View style={{flex: 1, flexDirection: 'row'}}>   
                 <InputCustom
                    placeholder="i.e. PA"
                    label="State: "
                    value={props.state}
                    onChangeText={props.onStateChge}
                    error={props.errorState}
                    inputStyle = {CustomInputStyles.inputStyleOsLeft}
                    containerStyle = {CustomInputStyles.containerStyleLeft}      
                    labelStyle = {CustomInputStyles.labelStyle}
                    errorStyle = {Platform.Os === 'ios' ? CustomInputStyles.error : CustomInputStyles.errorAndroid} 
                /> 
                <InputNumberCustom
                    placeholder="i.e. 19901"
                    label="Zip: "
                    value={props.zip}
                    onChangeText={props.onZipChge}
                    error={props.errorZip}
                    inputStyle = {CustomInputStyles.inputStyleOsRight }
                    containerStyle = {CustomInputStyles.containerStyleRight}      
                    labelStyle = {CustomInputStyles.labelStyleRight}
                    errorStyle = {Platform.Os === 'ios' ? CustomInputStyles.error : CustomInputStyles.errorAndroid} 
                />
            </View>
        </View>
    )
}

export default AccountAddress