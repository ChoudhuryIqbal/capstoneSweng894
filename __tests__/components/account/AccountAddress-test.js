import React from 'react'
import AccountAddress from '../../../src/components/account/AccountAddress'
import {create} from 'react-test-renderer';

describe("AccountAddress correctly", () => {
    test("AccountAddress renders without crashing", () => {
        const address = create(<AccountAddress />);
        expect (address.toJSON()).toBeTruthy();
    })
  });