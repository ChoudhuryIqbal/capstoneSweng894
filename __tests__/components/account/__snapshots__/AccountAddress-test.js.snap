// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountAddress correctly Matches the snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 10,
        "borderWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#ffffff",
          "display": "flex",
          "flex": 1,
          "flexBasis": 90,
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 16,
            "paddingBottom": 10,
          }
        }
      >
        Address: 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        placeholder="i.e. 100 Main Street"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "color": "#000",
            "elevation": 1,
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 3,
            "paddingBottom": 20,
            "paddingLeft": 5,
            "paddingRight": 5,
            "shadowOffset": Object {
              "height": 10,
              "width": 5,
            },
            "shadowOpacity": 1,
            "shadowRadius": 5,
          }
        }
        underlineColorAndroid="transparent"
        underlineColorAnroid="transparent"
      />
      <Text
        style={
          Object {
            "bottom": 0,
            "color": "red",
            "elevation": 1,
            "fontSize": 10,
            "justifyContent": "center",
            "marginTop": 10,
            "position": "absolute",
            "textAlign": "center",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#ffffff",
          "display": "flex",
          "flex": 1,
          "flexBasis": 90,
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 16,
            "paddingBottom": 10,
          }
        }
      >
        City: 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        placeholder="i.e. Malven"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "color": "#000",
            "elevation": 1,
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 3,
            "paddingBottom": 20,
            "paddingLeft": 5,
            "paddingRight": 5,
            "shadowOffset": Object {
              "height": 10,
              "width": 5,
            },
            "shadowOpacity": 1,
            "shadowRadius": 5,
          }
        }
        underlineColorAndroid="transparent"
        underlineColorAnroid="transparent"
      />
      <Text
        style={
          Object {
            "bottom": 0,
            "color": "red",
            "elevation": 1,
            "fontSize": 10,
            "justifyContent": "center",
            "marginTop": 10,
            "position": "absolute",
            "textAlign": "center",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 10,
        "borderWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#ffffff",
          "display": "flex",
          "flex": 1,
          "flexBasis": 90,
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 16,
            "paddingBottom": 10,
          }
        }
      >
        State: 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        placeholder="i.e. PA"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "color": "#000",
            "elevation": 1,
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 3,
            "paddingBottom": 20,
            "paddingLeft": 5,
            "paddingRight": 5,
            "shadowOffset": Object {
              "height": 10,
              "width": 5,
            },
            "shadowOpacity": 1,
            "shadowRadius": 5,
          }
        }
        underlineColorAndroid="transparent"
        underlineColorAnroid="transparent"
      />
      <Text
        style={
          Object {
            "bottom": 0,
            "color": "red",
            "elevation": 1,
            "fontSize": 10,
            "justifyContent": "center",
            "marginTop": 10,
            "position": "absolute",
            "textAlign": "center",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#ffffff",
          "display": "flex",
          "flex": 1,
          "flexBasis": 90,
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 16,
            "paddingBottom": 10,
          }
        }
      >
        Zip: 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        keyboardType="numbers-and-punctuation"
        placeholder="i.e. 19901"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "color": "#000",
            "elevation": 1,
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 3,
            "paddingBottom": 20,
            "paddingLeft": 5,
            "paddingRight": 5,
            "shadowOffset": Object {
              "height": 10,
              "width": 5,
            },
            "shadowOpacity": 1,
            "shadowRadius": 5,
          }
        }
        underlineColorAndroid="transparent"
        underlineColorAnroid="transparent"
      />
      <Text
        style={
          Object {
            "bottom": 0,
            "color": "red",
            "elevation": 1,
            "fontSize": 10,
            "justifyContent": "center",
            "marginTop": 10,
            "position": "absolute",
            "textAlign": "center",
          }
        }
      />
    </View>
  </View>
</View>
`;
